examples/hello-world/step-by-step/higgs/sklearn-linear/sklearn_linear.ipynb

```diff
@@ -0,0 +1,5516 @@
+    "    site_name = flare.get_site_name()\n",
+    "    \n",
+    "```\n",
+    "    \n",
+    "These few lines, import NVFLARE Client API and initialize it, then use the API to find the site_name (such as site-1, site-2 etc.). With the site-name, we can construct the site-specific \n",
+    "data path such as\n",
+    "\n",
+    "```\n",
+    "    feature_data_path = f\"{data_root_dir}/{site_name}_header.csv\"\n",
+    "\n",
+    "    data_path = f\"{data_root_dir}/{site_name}.csv\"\n",
+    "```\n",
+    "\n",
+    "#### Training \n",
+    "\n",
+    "In the standard traditional scikit learn, we would construct the model such as\n",
+    "```\n",
+    "  model = SGDClassifier(...) \n",
+    "```\n",
+    "then call model.fit(...)\n",
+    "```\n",
+    "  model.fit(x_train, y_train)\n",
+    "\n",
+    "  accuracy, report = evaluate_model(x_test, model, y_test)\n",
+    "\n",
+    "```\n",
+    "\n",
+    "with federated learning, using FLARE Client API, we have to make a few changes\n",
+    "* 1) we are not only training in local iterations, but also global rounds, we need to keep the program running until we reached to the totoal number of rounds \n",
+    "  \n",
+    "  ```\n",
+    "      while flare.is_running():\n",
+    "          ... rest of code\n",
+    "  \n",
+    "  ```\n",
+    "  \n",
+    "* 2) Unlike traditional machine learning, we have now have more than one clients/sites participating the training. To ensure every site starts with the same model parameters, we use server to broadcase the initial model parameters to every sites at the first round ( current_round = 0). \n",
+    "\n",
+    "* 3) We will need to use FLARE client API to receive global model and find out the global parameters\n",
+    "\n",
+    "```\n",
+    "     # (3) receives FLModel from NVFlare\n",
+    "        input_model = flare.receive()\n",
+    "        global_params = input_model.params\n",
+    "        curr_round = input_model.current_round\n",
+    "```\n",
+    "\n",
+    "```\n",
+    "      if curr_round == 0:\n",
+    "            # (4) initialize model with global_param\n",
+    "            # and set to all zero\n",
+    "            fit_intercept = bool(global_params[\"fit_intercept\"])\n",
+    "            model = SGDClassifier(\n",
+    "                loss=global_params[\"loss\"],\n",
+    "                penalty=global_params[\"penalty\"],\n",
+    "                fit_intercept=fit_intercept,\n",
+    "                learning_rate=global_params[\"learning_rate\"],\n",
+    "                eta0=global_params[\"eta0\"],\n",
+    "                max_iter=1,\n",
+    "                warm_start=True,\n",
+    "                random_state=random_state,\n",
+    "            )\n",
+    "        ....\n",
+    "```\n",
+    "* 4) if it is not the first round, we need to use the global model to update the local model before training the next round. For Scikit-learn leaner SDG model, we simply update coeffient and intercept. \n",
```

Comment:
typo: "For Scikit-learn leaner SDG model" -> "For Scikit-learn SGDClassifier's model"